src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -352,18 +390,24 @@ def test_WINNF_FT_S_FAD_1(self, config_filename):
         self.assertEqual(ppa_record['id'].split("/")[0], 'zone')
         self.assertEqual(ppa_record['id'].split("/")[1], 'ppa')
         self.assertEqual(ppa_record['id'].split("/")[2], self._sas._sas_admin_id)
-        del ppa_record['id']
-        # TODO : to update the following check according when PAL grants are supported in this test case
-        if 'cbsdReferenceId' in ppa_record['ppaInfo']:
-          self.assertFalse(ppa_record['ppaInfo']['cbsdReferenceId'])
-          del ppa_record['ppaInfo']['cbsdReferenceId']
+        del ppa_record['id'] 
         # verify that the injected ppas exist in the dump files
-        # TODO: add Check for GeoJSON Winding of ppa polygon  
+        # check GeoJson Winding of PPA record
+        utils.HasCorrectGeoJsonWinding(ppa_record['zone']['features'][0]['geometry'])     
         exist_in_dump = False
         for ppa in config['ppaRecords']:
           if 'id' in ppa:
-            del ppa['id']        
-          exist_in_dump = exist_in_dump or areTwoPpasEqual(ppa_record, ppa)
+            del ppa['id'] 
+            # we exclude the values of the CBSD reference Id, because they are not necessarily the same
+            ppa_cbsd_reference_ids = ppa['cbsdReferenceId']
+            ppa_record_cbsd_reference_ids = ppa_record['cbsdReferenceId']
+            del ppa['cbsdReferenceId']
+            del ppa_record['cbsdReferenceId']
+            exist_in_dump = exist_in_dump or areTwoPpasEqual(ppa_record, ppa)
+            # check that the record has the same length of the CBSD reference Ids
+            exist_in_dump &= len(ppa_cbsd_reference_ids) == len(ppa_record_cbsd_reference_ids)
```

Comment:
You should be able to check the values of the CBSD reference IDs as well. They are deterministically generated based on the FCC ID and SN.

Reply:
in admin testing API doc there is a note for inject zone(PPA) : the cbsdReferenceId field in the PpaInformation object provided through this API will contain CBSD IDs, not the corresponding CBSD Reference IDs as defined in the SAS-SAS TS. Depending on the SAS's implementation, these may or may not have the same format.

is the note is still relevant to this issue? or should SAS UUT set the CBSD reference ID in the dump the same as the injected one with the PPA?

Reply:
That note is still accurate. I think what you need to do is create each PPA's cluster list *including the CBSD reference IDs*. Right now we have the index of each CBSD, so you just have to pull the right reg request, get the FCC ID and serial number, then use a helper function to generate the CBSD reference ID from the FCC ID, SN. Then you will have the same format that the SAS is expected to provide and I think your existing comparison function will work correctly.